src/transformState.js

```diff
@@ -4,8 +4,72 @@
  * @param {Object} state
  * @param {Object[]} actions
  */
+
+// The function returns a new copy of the object with changes
 function transformState(state, actions) {
-  // write code here
+  if (isObject(state)) {
+    throw new Error('State is not an object');
+  }
+
+  if (isObject(actions)) {
+    throw new Error('Actions is not an object');
+  }
+
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (type) {
+      case 'addProperties':
+        const { extraData } = action;
```

Comment:
But why?

Reply:
It is better to use destructuring to obtain several values ​​from an object. If we need only one value, we can directly take it from the object.
it will just be cleaner and have less code. But this is just a recommendation.